Riot/AppDelegate.m

```diff
@@ -1176,6 +1276,54 @@ - (void)application:(UIApplication *)application handleActionWithIdentifier:(NSS
     completionHandler();
 }
 
+// iOS 10+, see application:didReceiveLocalNotification:
+- (void)userNotificationCenter:(UNUserNotificationCenter *)center willPresentNotification:(UNNotification *)notification withCompletionHandler:(void (^)(UNNotificationPresentationOptions options))completionHandler
+{
+    NSLog(@"[AppDelegate][Push] willPresentNotification: applicationState: %@", @([UIApplication sharedApplication].applicationState));
+
```

Comment:
Missing call to completion handler here. As we currently do nothing when we receive a notification when the application is in foreground we should use `completionHandler(UNNotificationPresentationOptionNone);`. Indeed this delegate method is called when a notification is delivered to a foreground app. 

We could remove all code related to room navigation and just keep this line `completionHandler(UNNotificationPresentationOptionNone);`. There is no reason to navigate to a room each time we receive a notification in foreground.
Instead we should handle navigation to a room when user tap the notification in `userNotificationCenter: didReceiveNotificationResponse:withCompletionHandler:` as default action.

Reply:
https://github.com/vector-im/riot-ios/pull/2207/commits/7ae3070e0ae39b0ff50c33f11ac7597e8e3ad50c